scripts/test.mk

```diff
@@ -0,0 +1,52 @@
+# This file is part of the repository template project. Please, DO NOT edit this file.
+
+# The test types listed here are both those which might run both locally and in CI, or
+# in one but not the other.  All of the test types listed at
+# https://github.com/NHSDigital/software-engineering-quality-framework/blob/main/quality-checks.md
+# should be represented here with the exception of:
+#   - dependency scanning, which we expect to be applied at the repository level
+#   - secret scanning, which we expect to be a pre-commit hook
+#   - code review, which is outside the scope of automated testing for the moment
+
+test-integration:
+	@make _test name="integration"
+
+test-contract:
+	@make _test name="contract"
+
+test-ui:
+	@make _test name="ui"
+
+test-accessibility:
+	@make _test name="accessibility"
+
+test-lint:
+	@make _test name="lint"
+
+# test-code-quality covers checking for duplicate code, code smells, and dead code.
+test-code-quality:
+	@make _test name="code-quality"
+
+test-unit:
+	@make _test name="unit"
+
+test-coverage:
+	@make -test name="coverage"
+
+test: test-unit \
+	test-lint \
+	test-code-quality \
+	test-coverage \
+	test-contract \
+	test-ui \
+	test-integration \
+	test-accessibility
+
+_test:
+	@set -e ;\
```

Comment:
That doesn't work for me.
```
 $ make --version
GNU Make 3.81
Copyright (C) 2006  Free Software Foundation, Inc.
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.

This program built for i386-apple-darwin11.3.0
```

That's the `/usr/bin/make` I've got by default on OS X.

Reply:
I might accept "here is a Linux laptop for you" as a resolution here 😂

Reply:
The solution to that is to instal the GNU toolchain for macOS. This is a necessity if our intention is to use shell scripts that are executable on Linux, WSL and on macOS. Please, take a look at this

- [Packages to be installed on macOS](https://github.com/nhs-england-tools/dotfiles/blob/6237ac323fddab2695232171a22fd82ad02fe2fe/assets/02-install-base-packages.macos.sh#L28) not all of them are required but recommended; `make` is on the list
- [Configuration to be added to `PATH`](https://github.com/nhs-england-tools/dotfiles/blob/6237ac323fddab2695232171a22fd82ad02fe2fe/dot_path.tmpl#L6) for both `amd64` and `arm` architecture

Reply:
Are we saying that `dotfiles` is non-optional?

Reply:
It's optional. However, we don't want to reinvent the wheel and solve a problem that the industry has already addressed. The dotfiles project includes an implementation of the industry-recommended solution to that particular problem (as shown in the code snippet I linked above), among others. Whether engineers choose to use this solution, another one, or something else, we can leave that up to them. The key point here is that we have resources available to guide them to the right solutions. In this way we can avoid a legacy way of implementing make recipes, or anything else.

To solve your problem in the simplest way possible, please try the following:

```
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
brew install make
```

I would say that the above is essential for a developer's workstation..

Reply:
I've got homebrew installed already, that's not the issue I'm trying to get at.  The README says that GNU Make is a dependency.  I have GNU Make installed.  The problem here is that we've got an undeclared dependency.  

There is nothing *in this repository specifically* to direct people at `dotfiles`, or to indicate that there's anything I would need to do in addition to just running `make` to get things to work.  In fact, if I do that and run into the inevitable errors that the use of `.ONESHELL` in `init.mk` will cause, my immediate assumption would be that the `Makefile` is simply broken.

To be absolutely clear here, I *do not recognise* the need for escaped newlines as a problem at all.  It's a choice.  While ungainly, it would be perfectly possible to have "shell scripts that are executable on Linux, WSL and on macOS" by targeting the lowest common denominator and pretending `.ONESHELL` doesn't exist.  Making a policy choice which turns that into a problem is a perfectly reasonable one to make, but it's not the default so it needs to be communicated.

Reply:
Thanks @regularfry, this PR looks good 👍, just about to merge it